lib/kubeclient/common.rb

```diff
@@ -51,6 +53,9 @@ def initialize_client(
       validate_auth_options(auth_options)
       handle_uri(uri, path)
 
+      @class_owner = class_owner
+      @entities = nil
+      @discovered = false
```

Comment:
I noticed in case the returned APIResourceList is empty or has only things that aren't parsed (like nested resources) that could cause an infinite loop in method_missing. 
I would like to have some guard against that